OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -257,22 +257,24 @@ protected void ReserveSpawnBuilding()
 			if (afld == null)
 				return;
 
-			var res = afld.TraitOrDefault<Reservable>();
+			MakeReservation(afld);
+		}
 
-			if (res != null)
-			{
-				UnReserve();
-				Reservation = res.Reserve(afld, self, this);
-			}
+		public void MakeReservation(Actor target)
```

Comment:
`Reservable.Reserve` never fails. If you call it on an already reserved target the old reservation is simply overwritten without warning - this probably leads to bad things but that's how it currently works.

All the current callers use `Reservable.IsReserved` to filter out already reserved locations before trying to make a reservation.